source/operand.cpp

```diff
@@ -29,1066 +29,21 @@
-    {SPV_OPERAND_TYPE_IMAGE_CHANNEL_ORDER, ARRAY_SIZE(imageChannelOrderEntries),
-     imageChannelOrderEntries},
-    {SPV_OPERAND_TYPE_IMAGE_CHANNEL_DATA_TYPE,
-     ARRAY_SIZE(imageChannelDataTypeEntries), imageChannelDataTypeEntries},
-    {SPV_OPERAND_TYPE_IMAGE, ARRAY_SIZE(imageOperandEntries),
-     imageOperandEntries},
-    {SPV_OPERAND_TYPE_OPTIONAL_IMAGE,  // Same as *_IMAGE
-     ARRAY_SIZE(imageOperandEntries), imageOperandEntries},
-    {SPV_OPERAND_TYPE_FP_FAST_MATH_MODE, ARRAY_SIZE(fpFastMathModeEntries),
-     fpFastMathModeEntries},
-    {SPV_OPERAND_TYPE_FP_ROUNDING_MODE, ARRAY_SIZE(fpRoundingModeEntries),
-     fpRoundingModeEntries},
-    {SPV_OPERAND_TYPE_LINKAGE_TYPE, ARRAY_SIZE(linkageTypeEntries),
-     linkageTypeEntries},
-    {SPV_OPERAND_TYPE_ACCESS_QUALIFIER, ARRAY_SIZE(accessQualifierEntries),
-     accessQualifierEntries},
-    {SPV_OPERAND_TYPE_OPTIONAL_ACCESS_QUALIFIER,  // Same as *_ACCESS_QUALIFIER
-     ARRAY_SIZE(accessQualifierEntries), accessQualifierEntries},
-    {SPV_OPERAND_TYPE_FUNCTION_PARAMETER_ATTRIBUTE,
-     ARRAY_SIZE(functionParameterAttributeEntries),
-     functionParameterAttributeEntries},
-    {SPV_OPERAND_TYPE_DECORATION, ARRAY_SIZE(decorationEntries),
-     decorationEntries},
-    {SPV_OPERAND_TYPE_BUILT_IN, ARRAY_SIZE(builtInEntries), builtInEntries},
-    {SPV_OPERAND_TYPE_SELECTION_CONTROL, ARRAY_SIZE(selectionControlEntries),
-     selectionControlEntries},
-    {SPV_OPERAND_TYPE_LOOP_CONTROL, ARRAY_SIZE(loopControlEntries),
-     loopControlEntries},
-    {SPV_OPERAND_TYPE_FUNCTION_CONTROL, ARRAY_SIZE(functionControlEntries),
-     functionControlEntries},
-    {SPV_OPERAND_TYPE_MEMORY_SEMANTICS_ID, ARRAY_SIZE(memorySemanticsEntries),
-     memorySemanticsEntries},
-    {SPV_OPERAND_TYPE_MEMORY_ACCESS, ARRAY_SIZE(memoryAccessEntries),
-     memoryAccessEntries},
-    {SPV_OPERAND_TYPE_OPTIONAL_MEMORY_ACCESS,  // Same as *_MEMORY_ACCESS
-     ARRAY_SIZE(memoryAccessEntries), memoryAccessEntries},
-    {SPV_OPERAND_TYPE_SCOPE_ID, ARRAY_SIZE(scopeEntries), scopeEntries},
-    {SPV_OPERAND_TYPE_GROUP_OPERATION, ARRAY_SIZE(groupOperationEntries),
-     groupOperationEntries},
-    {SPV_OPERAND_TYPE_KERNEL_ENQ_FLAGS,
-     ARRAY_SIZE(kernelKernelEnqueueFlagssEntries),
-     kernelKernelEnqueueFlagssEntries},
-    {SPV_OPERAND_TYPE_KERNEL_PROFILING_INFO,
-     ARRAY_SIZE(kernelProfilingInfoEntries), kernelProfilingInfoEntries},
-    {SPV_OPERAND_TYPE_CAPABILITY, ARRAY_SIZE(capabilityInfoEntries),
-     capabilityInfoEntries},
-};
+// Pull in operand info tables automatically generated from JSON grammar.
+#include "operand.inc"
 
 spv_result_t spvOperandTableGet(spv_operand_table* pOperandTable) {
   if (!pOperandTable) return SPV_ERROR_INVALID_POINTER;
 
-  static const spv_operand_table_t table = {ARRAY_SIZE(opcodeEntryTypes),
-                                            opcodeEntryTypes};
+  static const spv_operand_table_t table = {
+      ARRAY_SIZE(_pygen_variable_OperandInfoTable),
```

Comment:
Global scope identifiers with a single leading underscore are reserved for system use.